### ISSUE TITLE
Proposal for Auto-Activate New speaker function

### ISSUE THREAD

**Issue by LordMyschkin** [ejurgensen/forked-daapd#518]
I just want to propose a feature maybe not only I do really miss: When playing music and changing the room, I first switch on the speaker in the new room, then open whatever remote I use and activate the newly found speaker.
Would‘nt it be generally convenient if - when playing music by forked-daapd at the moment - the speaker could be activated as soon as it is detected in the network? Maybe as an option in forked-daapd.conf?

**Comment by ejurgensen** [comment]
Very good suggestion, I could also make use of that myself. I will look into it - hope it can be done easily.

**Comment by ejurgensen** [comment]
I made an attempt at the change in the above, but in my setup it doesn't work. It seems forked-daapd ends up attempting to connect too quickly, and the speaker returns "500 Internal server error". I also see forked-daapd sometimes crashing, because during startup the speaker quickly (mdns-wise) appears, disappears and then permanently reappears again. Forked-daapd tries to connect already at the first appearance, and it then crashes while it is trying to connect to a device that briefly vanishes.

So it wasn't as easy as I hoped...

**Comment by LordMyschkin** [comment]
I just want to tell you that I solved the problem for me by a small python script running in the backgound scanning for new speakers and activating them if forked-daapd is playing at that moment.
I am quite happy with the result, running on a raspberry3.
[enable_speakers.zip](https://github.com/ejurgensen/forked-daapd/files/2280566/enable_speakers.zip)

**Comment by ejurgensen** [comment]
Thanks, good to hear. Sorry I wasn't able to build this in.

**Comment by LordMyschkin** [comment]
Thank you for looking into this, anyway. May we close or should I leave that one open?

**Comment by ejurgensen** [comment]
Let's keep it open, maybe I can get it done some other time.

**Comment by jshep321** [comment]
Good idea, but I think this could be a bad idea to enable by default.  If speaker clients or wifi router/repeaters start dropping out temporarily, having music automagically start in certain rooms could lead to undesired behavior and quite negative... uh... marital discourse.
